AutoDraft/NHL_API.py

```diff
@@ -0,0 +1,499 @@
+import json, time
+import streamlit as st
+import numpy as np 
+import pandas as pd
+from pandas.io.json import json_normalize
+from datetime import datetime as dt
+import requests as rqsts  
+from bokeh.layouts import gridplot, column
+from bokeh.models import ColumnDataSource, RangeTool, Span, HoverTool
+from bokeh.models.glyphs import Patch
+from bokeh.plotting import figure, show
+import pmdarima as pm
+from sklearn.preprocessing import PowerTransformer
+
+## data ingestion
+def get_seasons(): # returns all seasons on record
+    seasons_response = rqsts.get('https://statsapi.web.nhl.com/api/v1/seasons')
+    if seasons_response.status_code != 200:
+        st.error('Failed attempt to get list of seasons.')
+        return
+    seasons = seasons_response.content
+    seasons_df = pd.read_json(seasons)
+    seasons_df = json_normalize(seasons_df.seasons)
+    seasons_df.set_index('seasonId', inplace=True)
+    return seasons_df
+
+def get_teams(): # returns all teams FOR THE CURRENT SEASON
+    teams_response = rqsts.get('https://statsapi.web.nhl.com/api/v1/teams')
+    if teams_response.status_code != 200:
+        st.error('Failed attempt to get list of teams.')
+        return
```

Comment:
may want to `raise` and exception rather than `return`. and then try/except this function call. or use the responses library's [`raise_for_status` method](https://2.python-requests.org//en/latest/user/quickstart/#response-status-codes). same comment below for other non-200 responses.

Reply:
Changed to use `{}.raise_for_status()`.